http/java/okhttp/lib/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -180,32 +181,68 @@ private String getMediaTypeAndSubType(final MediaType mediaType) {
                     response.close();
                 }
             } else {
-                return responseHandler.handleResponseAsync(response);
+                return responseHandler.handleResponseAsync(response, errorMappings);
             }
         });
     }
-    public <ModelType> CompletableFuture<Iterable<ModelType>> sendPrimitiveCollectionAsync(@Nonnull final RequestInformation requestInfo, @Nonnull final Class<ModelType> targetClass, @Nullable final ResponseHandler responseHandler) {
+    public <ModelType> CompletableFuture<Iterable<ModelType>> sendPrimitiveCollectionAsync(@Nonnull final RequestInformation requestInfo, @Nonnull final Class<ModelType> targetClass, @Nullable final ResponseHandler responseHandler, @Nullable final HashMap<String, Class<? extends Parsable>> errorMappings) {
         Objects.requireNonNull(requestInfo, "parameter requestInfo cannot be null");
 
-        return this.getHttpResponseMessage(requestInfo).thenCompose(response -> {
+        return this.getHttpResponseMessage(requestInfo)
+        .thenCompose(r -> this.throwFailedResponse(r, errorMappings))
```

Comment:
this bypasses the native response handler in case of an error, which is not what we're doing in other languages. Correct